### ISSUE TITLE
Slow backup network performance?

### ISSUE THREAD

**Issue by alphaleonis** [conformal/cyphertite#36]
I just signed up for Cyphertite and attempted a couple of backups. But the upload speed seems to be capped at 1Mbps? (I have a 10MBps upload capability)
If I run two backups simultaneously, each one is uploading at 1MBps it seems. I noted in another post on the old forum that upload speeds should be higher than this. Any idea of what's going on here?

I'm running Windows 8 x64 with Cypherite 1.6.2.

**Comment by csmiken** [comment]
This is strange behavior, for sure... We have done quite a bit of testing with Cyphertite and when not limited by the network connection we have seen speeds from a Windows machine near 25Mbps. That said, I am running a test on Win 8 x64 with CT 1.6.2 as I write this and am averaging 5.8Mbps backing up over a Comcast ISP connection advertised at 10 up, 30 down. You for sure should get more speed than you see as stated above.

Is there any other information you can give so I can try to reproduce the issue? What are your system specifications (processor, RAM, hd)? Are there any other applications running in the background that may be affecting system performance? Wired or wireless network connection? Are you using the CLI or GUI to make these backups?

**Comment by alphaleonis** [comment]
I'm running an Intel i7 with 16GB RAM, so that is definitely not a limiting factor. And it really does seem like they are capped at 1MBps. I have a wired connection to my router, and no network speed problems with any other application. And the strange thing is that if I launch multiple cyphertite processes simultaneously, _each one_ is uploading at 1MBps, so running three processes the total is 3MBps, as can be seen in the screenshots below:

![cyphertite-resmon](https://f.cloud.github.com/assets/5114354/887265/01c061a0-f9f9-11e2-9768-c34148fea1ca.png)

![cyphertite-taskman](https://f.cloud.github.com/assets/5114354/887348/2597704a-f9fa-11e2-9379-1a22a2dd2789.png)

I have tried both the web-based windows service as well as the console application, with the same results. (Can't run multiple simultaneous backups with the windows service of course). Can't think of any application that is running that should be affecting this. To verify I ran it on another computer of mine running Windows Server 2012 with the same result, it uploads at 1MBps, pretty much exactly.

Any suggestions would be greatly appreciated.

Ouput of cyphertite.exe -V:
cyphertite version: 1.6.2
Run-time versions:
 assl: 1.4.1
 clens: 0.0.6
 clog: 0.6.4
 curl: libcurl/7.25.0 OpenSSL/1.0.1c zlib/1.2.5
 event: 2.0.17-stable
 exude: 0.7.3
 openssl: OpenSSL 1.0.1c 10 May 2012
 shrink: 0.5.4
 xmlsd: 0.10.1
O/S identification: Microsoft Professional-amd64-6.2 (build 9200)

**Comment by csmiken** [comment]
Thanks for the additional info. I spoke with one of my developers and he saw something interesting in your screen shot. The disk usage appears capped along with your network transfer rate. So this makes us think: is this a VSS issue? Would it be possible for you to start a backup of new data (to avoid deduplication) from a non-admin command prompt? A non-admin command prompt would prevent Cyphertite from using VSS and I am curious if the speed results would change, or be the same. Also, can you start a similar backup of new data from the GUI and let me know what ctd.exe does as far as network transfer and disk usage? Your machine is very much powerful enough as to not hit any bottlenecks that we can think of. We have not built in any caps to the client software and I am unable to reproduce it on my very similar setup here (win 8 64 bit i7 with 8GB RAM). So I appreciate your help to figure out what is going on...

**Comment by alphaleonis** [comment]
Okay, so I restarted some of my old backups from the CLI interface. One of the three I started reported the following:

[vssrl] src/vss_requester_template.cpp:612:InitializeVss(): Unable to create VSS backup components.
snapshotting will not be used due to insufficient credentials

which I guess is what we wanted (even though it was from an administrator prompt, so a bit strange, but may have something to do with the fact that multiple VSS requests were made simultaneously or something. VSS has a habit of being a bit dodgy some times if not everything is done correctly). (I am a software developer myself that wrote the AlphaVSS library, so have some experience with these issues).

Either way, I also started a new backup from a non-admin account, and got the same warning as above. 

The backups that were restarted showed initially high disk-usage (25-60MB/s), but once they started uploading the network speed was back at 1MBps, and the disk usage at about 0,1-0,2MB/s. The same is true for the GUI-interface, ctd.exe, with a new backup 1MBps network and 0,1-0,2MB/s disk usage.

So it does not seem to be a VSS issue as far as I can tell, because of well, both the high initial disk speed, and the fact that the results are the same whether VSS is used or not.  

The other strange thing is that the same behavior occurred from my other computer, which incidentally had a brand new installation of Windows 2012 Server on it, and nothing else installed at all.

In conclusion all I can say is that the speed is limited to the numbers I've mentioned no matter what I do with respect to admin/non-admin, VSS/no-vss, new backup or resumed backup.

**Comment by csmiken** [comment]
Alright, thanks for the info. You are right, this does eliminate VSS as a cause. I will chat with the guys for more ideas. Now I am still very confused as to why you are stuck at 1Mb while mine saturates my upload speed. Would you mind emailing your Cyphertite username to support@cyphertite.com so I can take a look at your account? I am interested to see if there is any reason our back end might be treating you differently (not by design). I will update again here once I have a chat with the guys.

**Comment by alphaleonis** [comment]
As a coincidence I actually swapped internet providers today, where I went from cable to fiber, still a 100/10MBit/s connection. Also, I connected the computer directly to the wall outlet bypassing any router or similar, and the speeds are the same, so that eliminates that part of the network as the source of the problem anyway.

**Comment by csmiken** [comment]
Confirmed by a second user. I have asked for some additional information. We are still unable to re-create this behavior on any of our test machines. Hopefully some new data will help us sort this out.

**Comment by csmiken** [comment]
Good news! After much testing and some data gathering from a few users in different physical locations we have identified the cause of this speed issue that has been seen by a small number of our users. We have implemented a fix server-side that should improve the speed on Windows OSes by up to 4x. There will be a release soon to obtain a similar speed increase for other OSes. Keep in mind: this slowness was seen by a limited amount of users. Not all users will see an increase, because the majority of our users were not affected with the bottleneck.

**Comment by alphaleonis** [comment]
This problem seems to be solved in version 2.0 for windows anyway. Thanks.
